hudi-client/src/main/java/org/apache/hudi/index/hbase/HBaseIndex.java

```diff
@@ -287,13 +289,10 @@ private boolean checkIfValidCommit(HoodieTableMetaClient metaClient, String comm
           hbaseConnection = getHBaseConnection();
         }
       }
-      HTable hTable = null;
-      try {
-        hTable = (HTable) hbaseConnection.getTable(TableName.valueOf(tableName));
+      try (BufferedMutator mutator = hbaseConnection.getBufferedMutator(TableName.valueOf(tableName))) {
```

Comment:
Could you comment with a small snippet of the doc mentioning this is the new API ?

Reply:
Sure. Here is the doc: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html.
```
The try-with-resources statement is a try statement that declares one or more resources.
A resource is an object that must be closed after the program is finished with it.
The try-with-resources statement ensures that each resource is closed at the end of the statement.
```

Reply:
Thanks for providing the link, so is the `BufferedMutator` from Hbase also the preferred new methodology ? (The above link tells me that from the JDBC driver changes are fair)

Reply:
Yes I think so. 
The HBase API doc shows:
```
flushCommits()
Deprecated. 
as of 1.0.0. Replaced by BufferedMutator.flush()
```
Also HTable is not threadsafe. (https://issues.apache.org/jira/browse/HBASE-17361)

Reply:
Ok great, thanks!